src/app/api/treasure-box/route.ts

```diff
@@ -133,14 +139,31 @@ export async function POST(request: NextRequest) {
     }
   }
 
+  let currentScore = score.current_score;
+  if (iykRef) {
+    console.log('got iykRef:', iykRef);
+    const result = await getTapRef(iykRef as string);
+    if (result === null || !result.isValidRef) {
```

Comment:
/nit checking for positive condition is easier to read and doesn't require else

```suggestion
    if (result && result.isValidRef) {
```